roles/redhat_packages/tasks/main.yml

```diff
@@ -0,0 +1,71 @@
+---
+
+- name: gather a list of all rpms
+  raw: rpm -qa --qf "%{NAME}|%{VERSION}|%{RELEASE}|%{INSTALLTIME}|%{VENDOR}|%{BUILDTIME}|%{BUILDHOST}|%{SOURCERPM}|%{LICENSE}|%{PACKAGER}|%{INSTALLTIME:date}|%{BUILDTIME:date}|%{DSAHEADER:pgpsig}|%{RSAHEADER:pgpsig}|%{SIGGPG:pgpsig}|%{SIGPGP:pgpsig}|\n"
+  register: redhat_packages_all
+  ignore_errors: yes
+  when: internal_have_rpm
+
+- name: set fact of all installed rpms
+  set_fact:
+    redhat_packages_all: "{{ redhat_packages_all }}"
+
+- name: gather a list of all red hat rpms
+  raw: rpm -qa --qf "%{NAME}|%{VERSION}|%{RELEASE}|%{INSTALLTIME}|%{VENDOR}|%{BUILDTIME}|%{BUILDHOST}|%{SOURCERPM}|%{LICENSE}|%{PACKAGER}|%{INSTALLTIME:date}|%{BUILDTIME:date}|%{DSAHEADER:pgpsig}|%{RSAHEADER:pgpsig}|%{SIGGPG:pgpsig}|%{SIGPGP:pgpsig}|\n" | grep 'Key ID 199e2f91fd431d51\|Key ID 5326810137017186\|Key ID 45689c882fa658e0\|Key ID 219180cddb42a60e\|Key ID 7514f77d8366b0d9\|Key ID 45689c882fa658e0'
+  register: redhat_packages_gpg_is_redhat
+  ignore_errors: yes
+  when: internal_have_rpm
+
+- name: set fact of all installed red hat rpms
+  set_fact:
+    redhat_packages_gpg_is_redhat: "{{ redhat_packages_gpg_is_redhat }}"
+
+- name: gather the number of all installed red hat packages filtered by gpg keys
+  raw: rpm -qa --qf "%{SIGPGP:pgpsig}\n" | grep 'Key ID 199e2f91fd431d51\|Key ID 5326810137017186\|Key ID 45689c882fa658e0\|Key ID 219180cddb42a60e\|Key ID 7514f77d8366b0d9\|Key ID 45689c882fa658e0' | wc -l
+  register: redhat_packages_gpg_num_rh_packages
+  ignore_errors: yes
+  when: internal_have_rpm
+
+- name: set fact of number of installed red hat packages filtered by gpg keys
+  set_fact:
+    redhat_packages_gpg_num_rh_packagespush: "{{ redhat_packages_gpg_num_rh_packages }}"
+
+- name: gather total number of installed packages
+  raw: rpm -qa | wc -l
+  register: redhat_packages_all_count
+  ignore_errors: yes
+  when: internal_have_rpm
+
+- name: set fact of number of all installed rpm packages
+  set_fact:
+    redhat_packages_gpg_num_installed_packages: "{{ redhat_packages_all_count }}"
+
+- name: gather the last installed red hat package filtered by gpg keys
+  raw: rpm -qa --qf "%{INSTALLTIME} - %{NAME}|%{VERSION}|%{RELEASE}|%{INSTALLTIME}|%{VENDOR}|%{BUILDTIME}|%{BUILDHOST}|%{SOURCERPM}|%{LICENSE}|%{PACKAGER}|%{INSTALLTIME:date}|%{BUILDTIME:date}|%{DSAHEADER:pgpsig}|%{RSAHEADER:pgpsig}|%{SIGGPG:pgpsig}|%{SIGPGP:pgpsig}|\n" | grep 'Key ID 199e2f91fd431d51\|Key ID 5326810137017186\|Key ID 45689c882fa658e0\|Key ID 219180cddb42a60e\|Key ID 7514f77d8366b0d9\|Key ID 45689c882fa658e0' | sort -nr | head -n 1
+  register: redhat_packages_gpg_last_installed
+  ignore_errors: yes
+  when: internal_have_rpm
+
+- name: set fact of last installed rh package filtered by gpg key
+  set_fact:
+    redhat_packages_gpg_last_installed: "{{ redhat_packages_gpg_last_installed }}"
```

Comment:
Do you not need to extract from `stdout` or `stdout_lines[0]`?  Could you add comment with the example output in the results from a scan?